stanford_soe_r25.module

```diff
@@ -0,0 +1,889 @@
+      }
+    }
+  }
+  cache_set($cache_id, $list);
+  return $list;
+}
+
+// This is where we hook into the reservation form to change things for
+// insidesoe field placement and overrides.
+function stanford_soe_r25_form_stanford_soe_r25_reservation_alter(&$form, &$form_state, $form_id) {
+  $form['stanford_soe_r25_booking_tentative']['#weight'] = -11;
+  $form['stanford_soe_r25_booking_date']['#weight'] = -10;
+  $form['stanford_soe_r25_booking_duration']['#weight'] = -9;
+  $form['stanford_soe_r25_booking_enddate']['#weight'] = -8;
+  $form['stanford_soe_r25_booking_reason']['#weight'] = -7;
+  $form['stanford_soe_r25_booking_attr191']['#weight'] = -3;
+  $form['stanford_soe_r25_booking_headcount']['#weight'] = -2;
+  $form['stanford_soe_r25_booking_attr144']['#weight'] = -1;
+  $form['stanford_soe_r25_booking_reason']['#title'] = 'Event Title';
+  $form['stanford_soe_r25_booking_attr182']['#title'] = 'Event Details';
+  $form['stanford_soe_r25_booking_attr191']['#title'] = 'Primary Organization';
+  $form['stanford_soe_r25_booking_attr141']['#type'] = 'radios';
+  $form['stanford_soe_r25_booking_attr141']['#options'] = array(1 => 'Yes', 0 => 'No');
+  $form['stanford_soe_r25_booking_attr141']['#default_value'] = variable_get('stanford_soe_r25_booking_attr141', 0);
+  $form['stanford_soe_r25_booking_attr166']['#type'] = 'radios';
+  $form['stanford_soe_r25_booking_attr166']['#options'] = array(1 => 'Yes', 0 => 'No');
+  $form['stanford_soe_r25_booking_attr166']['#default_value'] = variable_get('stanford_soe_r25_booking_attr141', 0);
+  $form['stanford_soe_r25_booking_attr178']['#suffix'] = '<div class="form-item"><span class="description">If unsure, please put N/A </span></div>';
+  $form['submit']['#prefix'] = '<div class="form-item"><span class="readandagree">By clicking the Request button, I confirm that I have read and agree to the <a href="https://insidesoe.stanford.edu/science-and-engineering-quad/policies-rates">Facility Rental Policy</a></span></div>';
+  $form['stanford_soe_r25_booking_attr144']['#suffix'] = '<div class="form-item"><span class="description">E.g., 1234567-123-ABCDE (Case sensitive)</span></div>';
+  $form['stanford_soe_r25_booking_attr198']['#suffix'] = '<div class="form-item"><span class="description">E.g. Reservation spans multiple days; Breakout spaces needed</span></div>';
+  $form['stanford_soe_r25_booking_date']['#suffix'] = '<div class="starttime">Start time includes set-up time</div>';
+  $form['stanford_soe_r25_booking_enddate']['#suffix'] = '<div class="endtime">End time includes break-down time</div>';
+  $form['stanford_soe_r25_booking_attr182']['#maxlength'] = 80;
+  array_unshift($form['#validate'], 'stanford_soe_r25_stanford_soe_r25_pta_validate');
+}
+
+function stanford_soe_r25_form_webform_client_form_564_alter(&$form, &$form_state, $form_id) {
+
+  $form['#node']->webform['components'][11]['type'] = 'textfield';
+  $form['#node']->webform['components'][13]['type'] = 'textfield';
```

Comment:
How do you know there are those components on any particular webform? why not just change those types in the UI?